fix: fix upgrade

niladrih opened the pull request:
- Fix preflight checks
- Change delete command to kubectl openebs upgrade delete
- Fix env from CORE_CHART_DIR to CHART_DIR
- Fix image tag to have dots instead of -

Abhinandan-Purkait commented on charts/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
-version: 4.3.0-develop
+version: 4.3.0
 name: openebs
-appVersion: 4.3.0-develop
+appVersion: 4.3.0
```
Why are we reverting this? The develop chart should always be vNext-develop

niladrih replied:
I was testing with these changes. I've reverted this change.

Abhinandan-Purkait commented on charts/Chart.yaml:
```diff
@@ -26,18 +26,18 @@ dependencies:
     condition: engines.local.lvm.enabled
   - name: mayastor
-    version: 0.0.0
+    version: 2.8.0
```
Same here as well, for all the dependents.

Abhinandan-Purkait commented on nix/pkgs/images/default.nix:
```diff
@@ -48,7 +47,7 @@ let
     if [[ "$(semver validate ${tag})" == "valid" ]] &&
       [[ ! ${tag} =~ ^(v?[0-9]+\.[0-9]+\.[0-9]+-0-(main|release)-unstable(-[0-9]+){6}-0)$ ]]; then
-      CHART_FILE=build/charts/Chart.yaml build/scripts/helm/update-chart-version.sh --app-version ${tag}
+      CHART_FILE=build/charts/Chart.yaml build/scripts/helm/update-chart-version.sh --app-version ${tag} --chart-version ${tag} --localpv-provisioner-version 4.2.0 --zfs-localpv-version 2.7.1 --lvm-localpv-version 1.6.2 --mayastor-version 2.8.0
```
Hmm, is this supposed to be hardcoded?

niladrih replied:
This needs some changes on update-version-on-release.sh. I've commented this block for now.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/cli.rs:
```diff
@@ -1,9 +1,16 @@
 use clap::{Parser, Subcommand};
 use plugin::ExecuteOperation;
 use std::path::PathBuf;
+use upgrade::common::kube::client::client;
```
Arrange the imports here.

niladrih replied:
I've arranged the imports.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/helm_values.rs:
```diff
@@ -0,0 +1,153 @@
+    replicated: ReplicatedEngines,
+}
+
+// #[derive(Deserialize, Debug)]
+// struct LocalEngines {
+//     lvm: EngineEnabled,
+//     zfs: EngineEnabled,
+// }
```
Commented code?

niladrih replied:
Removed!

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/helm_values.rs:
```diff
@@ -0,0 +1,153 @@
+
+#[derive(Deserialize, Debug)]
+struct Engines {
+    // local: LocalEngines,
```
commented code?

niladrih replied:
I've removed these.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/helm_values.rs:
```diff
@@ -0,0 +1,153 @@
+        self.chart.values.engines.replicated.mayastor.enabled
+    }
+
+    // /// Returns if the Local PV LVM storage engine is enabled in the helm values.
+    // pub fn localpv_lvm_is_enabled(&self) -> bool {
+    //     self.chart.values.engines.local.lvm.enabled
+    // }
+    //
+    // /// Returns if the Local PV ZFS storage engine is enabled in the helm values.
+    // pub fn localpv_zfs_is_enabled(&self) -> bool {
+    //     self.chart.values.engines.local.zfs.enabled
+    // }
```
?

niladrih replied:
Yep, removed.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/helm_values.rs:
```diff
@@ -0,0 +1,153 @@
+use crate::{cli_utils::upgrade::k8s::decode_decompress_data, extract_data};
+use anyhow::{anyhow, Result};
+use semver::Version;
+use serde::Deserialize;
+use upgrade::common::kube::client::{list_configmaps, list_secrets};
+
```
arrange the imports.

niladrih replied:
I've arranged the imports.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/k8s/mod.rs:
```diff
@@ -2,7 +2,7 @@ use crate::constants::{upgrade_obj_suffix, HTTP_DATA_PAGE_SIZE};
 use base64::engine::{general_purpose::STANDARD, Engine as base64_engine};
 use flate2::read::GzDecoder;
-use k8s_openapi::{api::events::v1::Event, kind};
+use k8s_openapi::api::{batch::v1::Job, events::v1::Event};
```
here as well

niladrih replied:
I've arranged the imports.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/mod.rs:
```diff
@@ -28,16 +31,21 @@ use k8s_openapi::{
 use tokio::{spawn, task::JoinHandle, try_join};
-use upgrade::common::kube::client::list_pods;
+use upgrade::common::kube::client::{list_pods, paginated_list_metadata};
+use utils::version_info;
```
Arrange the imports.

niladrih replied:
I've arranged the imports.

Abhinandan-Purkait commented on plugin/src/cli_utils/upgrade/mod.rs:
```diff
@@ -172,6 +180,7 @@ async fn upgrade_job(
                         args: Some(job_args),
                         image: Some(upgrade_image),
                         image_pull_policy: image_properties.pull_policy,
+                        // image_pull_policy: image_properties.pull_policy,
```
?

niladrih replied:
Testing artifact. Reverted this.

Abhinandan-Purkait commented on charts/shell.nix:
```diff
@@ -13,4 +13,4 @@ pkgs.mkShell {
     jq
   ];
-}
+}
```
Blank line

niladrih replied:
Reverted all changes here.

Abhinandan-Purkait commented on charts/values.yaml:
```diff
@@ -337,4 +337,4 @@ alloy:
         {{- end -}}
         {{- join "|" $regexParts -}}
-        {{- end -}}
+        {{- end -}}
```
here

niladrih replied:
Reverted this.

Abhinandan-Purkait commented on charts/charts/openebs-crds/values.yaml:
```diff
@@ -3,4 +3,4 @@ csi:
     enabled: true
     # -- Keep CRDs on chart uninstall
-    keep: true
+    keep: truen
```
keep: truen --> keep: true?

Also blank line

niladrih replied:
I've reverted all helm chart related changes.

Abhinandan-Purkait commented on charts/Chart.yaml:
```diff
@@ -122,4 +122,4 @@ annotations:
       image: registry.k8s.io/sig-storage/snapshot-controller:v6.2.2
     - name: snapshot-controller
-      image: registry.k8s.io/sig-storage/snapshot-controller:v6.3.3
+      image: registry.k8s.io/sig-storage/snapshot-controller:v6.3.3
```
here.

niladrih replied:
Reverted this.

Abhinandan-Purkait commented on charts/charts/openebs-crds/Chart.yaml:
```diff
@@ -1,4 +1,4 @@
 name: openebs-crds
 version: 4.3.0-develop
-description: A Helm chart that collects CustomResourceDefinitions (CRDs) from OpenEBS.
+description: A Helm chart that collects CustomResourceDefinitions (CRDs) from OpenEBS.
```
here

niladrih replied:
Reverted this.

niladrih committed:
fix: fix upgrade
- Fix preflight checks
- Change delete command to kubectl openebs upgrade delete
- Fix env from CORE_CHART_DIR to CHART_DIR
- Fix image tag to have dots instead of -

Signed-off-by: Niladri Halder <niladri.halder26@gmail.com>
